sentry_sdk/scope.py

```diff
@@ -134,10 +134,33 @@ def fingerprint(self, value):
         """When set this overrides the default fingerprint."""
         self._fingerprint = value
 
-    @_attr_setter
+    @property
+    def transaction(self):
+        # type: () -> Any
+        # would be type: () -> Optional[Span], see https://github.com/python/mypy/issues/3004
+        # XXX: update return type to Optional[Transaction]
+        """Return the transaction (root span) in the scope."""
+        if self._span is None or self._span._span_recorder is None:
+            return None
+        try:
+            return self._span._span_recorder.spans[0]
+        except (AttributeError, IndexError):
+            return None
+
+    @transaction.setter
     def transaction(self, value):
-        # type: (Optional[str]) -> None
+        # type: (Any) -> None
+        # would be type: (Optional[str]) -> None, see https://github.com/python/mypy/issues/3004
         """When set this forces a specific transaction name to be set."""
+        # XXX: the docstring above is misleading. The implementation of
```

Comment:
Feel free to fix that behavior in this PR, I don't think anybody would notice even.